assets/js/components/ErrorNotice.js

```diff
@@ -69,11 +70,12 @@ export default function ErrorNotice( {
 		hasButton && isErrorRetryable( error, selectorData );
 
 	// Append "Try again" messaging if no retry button is present.
-	if ( ! hasButton ) {
+	if ( ! hasButton && ! skipRetryMessage ) {
 		message = sprintf(
-			/* translators: %s: Error message from Google API. */
-			__( '%s. Please try again.', 'google-site-kit' ),
-			message
+			/* translators: %1$s: Error message from Google API. */
+			__( '%1$s%2$s Please try again.', 'google-site-kit' ),
+			message,
+			message.endsWith( '.' ) ? '' : '.'
```

Comment:
There was a bug with this component, where additional period was included even to the messages with period, so I fixed it here, as it is used for key metrics saving errors, and 2 dots (..) were appearing

Reply:
Good find, thank you